jbmc/src/java_bytecode/assignments_from_json.cpp

```diff
@@ -755,12 +755,7 @@ static get_or_create_reference_resultt get_or_create_reference(
       reference.array_length =
         info.allocate_objects.allocate_automatic_local_object(
           java_int_type(), "user_specified_array_length");
-      code.add(code_without_referencest{code_assignt{
-        *reference.array_length, side_effect_expr_nondett{java_int_type()}}});
-      code.add(code_without_referencest{code_assumet{binary_predicate_exprt{
-        *reference.array_length, ID_ge, from_integer(0, java_int_type())}}});
-      code.add(code_without_referencest{
-        allocate_array(reference.expr, *reference.array_length, info.loc)});
+      code.add(reference_allocationt{id, info.loc});
```

Comment:
Commit "Add reference allocation code for references" doesn't seem to do what it says (it only deletes code from `get_or_create_reference`)